sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -393,28 +393,35 @@ object DateTimeUtils {
   }
 
   /**
+   * Returns the microseconds since year zero (-17999).
+   */
+  def absoluteMicroSeconds(us: SQLTimestamp): SQLTimestamp = {
+    us + toYearZero * MICROS_PER_DAY
+  }
+
+  /**
    * Returns the hour value of a given timestamp value. The timestamp is expressed in microseconds.
    */
-  def getHours(timestamp: SQLTimestamp): Int = {
-    val localTs = (timestamp / 1000) + defaultTimeZone.getOffset(timestamp / 1000)
-    ((localTs / 1000 / 3600) % 24).toInt
+  def getHours(us: SQLTimestamp): Int = {
```

Comment:
There isn't that many. If you look at DateTimeUtils, all other units are spelled out, e.g. millis, nanoseconds.

Note that the problem is that the unit is not "u". It just looks like u...